firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md

```diff
@@ -0,0 +1,374 @@
+The `fs-bq-schema-views` script is for use with the official Firebase Extension
+[_Export Collections to BigQuery_](https://github.com/firebase/extensions/tree/master/firestore-bigquery-export).
+
+## Overview
+
+The `fs-bq-schema-views` script (referred to as the "schema-views script")
+generates richly-typed BigQuery views of your raw changelog.
+
+The _Export Collections to BigQuery_ extension only mirrors raw data, but it
+doesn't apply schemas or types. This decoupling makes schema validation less
+risky because no data can be lost due to schema mismatch or unknown fields.
+
+The schema-views script creates a BigQuery view, based on a JSON schema
+configuration file, using
+[BigQuery's built-in JSON functions](https://cloud.google.com/bigquery/docs/reference/standard-sql/json_functions).
+The _Export Collections to BigQuery_ extension also provides some BigQuery
+[user-defined functions](https://github.com/firebase/extensions/blob/master/firestore-bigquery-export/scripts/gen-schema-view/src/udf.ts)
+that are helpful in converting Firestore document properties to richly-typed
+BigQuery cells.
+
+## Use the script
+
+The following steps are an example of how to use the schema-views script. In the
+sections at the end of this file, you can review detailed information about
+configuring a schema file and reviewing the resulting schema views.
+
+### Step 1: Create a schema file
+
+The schema-views script runs against "schema files" which specify your schema
+configurations in a JSON format.  
+  
+In any directory, create a schema file called `test_schema.json`
+that contains the following:
+
+```
+{
+  "fields": [
+    {
+      "name": "name",
+      "type": "string"
+    },
+    {
+      "name": "age",
+      "type": "number"
+    }
+  ]
+}
+```
+
+Learn [How to configure schema files](#how-to-configure-schema-files)
+later in this guide.
+
+### Step 2: Set up credentials
+
+The schema-views script uses Application Default Credentials to communicate with
+BigQuery. 
+
+One way to set up these credentials is to run the following command using the
+[gcloud](https://cloud.google.com/sdk/gcloud/) CLI:
+
+```
+$ gcloud auth application-default login
+```
+
+Alternatively, you can
+[create and use a service account](https://cloud.google.com/docs/authentication/production#obtaining_and_providing_service_account_credentials_manually).
+This service account must be assigned a role that grants the permission of
+`bigquery.jobs.create`.
```

Comment:
Easier to say that they have to give it the role of "BigQuery Job User", so they don't have to look up which role has "bigquery.jobs.create" permission. 
(Docs are at https://cloud.google.com/iam/docs/understanding-roles if you're curious)

Reply:
Is this the most granular role available?  It looks like it is. @jhuleatt 

If it is, then we can say,
"... a role that grants the permission `bigquery.jobs.create`, like the ["BigQuery Job User" role](https://cloud.google.com/iam/docs/understanding-roles#bigquery-roles).

Reply:
done :-)